src/main/java/uk/gov/companieshouse/company/metrics/exception/ControllerExceptionHandler.java

```diff
@@ -130,6 +132,26 @@ public ResponseEntity<Object> handleMethodArgumentNotValidException(
         return new ResponseEntity<>(responseBody, HttpStatus.BAD_REQUEST);
     }
 
+    /**
+     * MethodNotAllowedException exception handler.
+     *
+     * @param ex      exception to handle.
+     * @param request request.
+     * @return error response to return.
+     */
+    @ExceptionHandler(value = {MethodNotAllowedException.class,
+            HttpRequestMethodNotSupportedException.class})
+    public ResponseEntity<Object> handleMethodNotAllowedException(Exception ex,
+                                                                  WebRequest request) {
+        var correlationId = generateShortCorrelationId();
+        logger.error(String.format("Started: handleException: %s Generating error response ",
+                correlationId), ex);
+        Map<String, Object> responseBody = new LinkedHashMap<>();
+        populateResponseBody(responseBody, correlationId);
+        return new ResponseEntity<>(responseBody, HttpStatus.METHOD_NOT_ALLOWED);
+
+    }
+
     private String generateShortCorrelationId() {
         return UUID.randomUUID().toString().replace(TARGET, REPLACEMENT).substring(0,8);
     }
```

Comment:
[out-of-scope, non-blocker] We had this months ago on the insolvency side and it served no purpose - random context id every time instead of the one we should be passing through is a bit weird isn't it.

Just flagging so it can be removed down the line.

Reply:
Thanks Chris